src/test/java/org/openrewrite/staticanalysis/AddStaticModifierToPublicFinalConstantsAndFieldsTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class AddStaticModifierToPublicFinalConstantsAndFieldsTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new AddStaticModifierToPublicFinalConstantsAndFields());
+    }
+
+    @Test
+    void doesNotModify() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class A {
+                  public final static String s1;
+                  public String s2;
+                  static String s3;
+              }
+              """
+          )
+        );
+    }
+
+    @DocumentExample
+    @Test
+    void addStaticModifier() {
```

Comment:
We might want to add another test here with an anonymous inner class, as I believe static fields are not allowed there, and we want to be sure that's handled correctly by the recipe to not introduce such modifiers there.